adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -38,488 +35,421 @@ import scala.reflect.ClassTag
 sealed abstract class ShuffleRegionJoin[T: ClassTag, U: ClassTag, RT, RU]
     extends RegionJoin[T, U, RT, RU] {
 
-  protected val sd: SequenceDictionary
-  protected val partitionSize: Long
-  protected val sc: SparkContext
+  protected def advanceCache(setTheoryCache: SetTheoryCache[U, RT, RU],
```

Comment:
Thanks for the ping and the detailed breakdown. On reflection, it would probably be clunky to move these methods onto `SetTheoryCache` (e.g., `pruneCache` taking 3 functions seems like a step backwards). Shall we keep them here?

> I had it as cache, but it looked weird to keep using cache.cache. I'll put it back as cache, I don't have a preference either way.

 I think it is slightly more readable as `cache`, but I don't have strong preferences.